modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/apps/edit/EditAppBody.es.js

```diff
@@ -56,12 +102,21 @@ export default ({defaultLanguageId, endpoint, title, ...restProps}) => {
 						defaultLanguageId={defaultLanguageId}
 						isEmpty={filteredItems.length === 0}
 						isLoading={isLoading}
-						items={filteredItems}
+						itemId={stepItems[itemType].id}
+						items={stepItems[itemType].items}
 						keywords={searchText}
+						onChange={(id) => {
+							dispatch({
+								id,
+								type: stepItems[itemType].type,
```

Comment:
Is it better we check if the item of the object is valid to prevent a javascript error? Something Like: `stepItems[itemType]?.id`

Reply:
This scenario is already covered by the object creation, the itemType value will not be a different key from the existing ones

Reply:
Ok, Thanks guys